### ISSUE TITLE
Question: What is the "Network" used for?

### ISSUE THREAD

**Issue by bazooka70** [elucidsoft/dotnet-stellar-sdk#108]
I see in the examples:

```csharp
// Set network and server
Network network = new Network("Test SDF Network ; September 2015");
Server server = new Server("https://horizon-testnet.stellar.org");
....
```
Or even:

```csharp
Network.UseTestNetwork();
Server server = new Server("https://horizon-testnet.stellar.org");
```

I went over the `Network` class and can't figure out what/where it is used for?
Since we instantiate the `Server` with specific uri (test or main), where does the `Network` used? I can't see its being used anywhere for example:
[examples/send_native_assets.md](https://github.com/elucidsoft/dotnet-stellar-sdk/blob/5abc95a7367e72a163906df5651a9fe77749f661/docfx/examples/send_native_assets.md)

Thanks :)

**Comment by Kirbyrawr** [comment]
Hi there i'm on my lunch time so i have time to reply to you.

A network is a cluster of servers connected between them in a decentralized way.

For connect to a network you need a passphrase.

If you ask the server it will give you the details of the network it's in, also it csn give you the passphrase https://horizon-testnet.stellar.org (try to click and see on the bottom).

Here is more info: https://www.stellar.org/how-it-works/stellar-basics/explainers/

It's a bit tricky though.

**Comment by bazooka70** [comment]
Thanks. so do I need to specify the `Network` before each request? (both test and main)
i would expect that the `new Server()` instance will take the `new Network()` instance as a parameter/property...

is using `Network.UseTestNetwork();` or `Network.UsePublicNetwork();` thread safe?

Again, sorry if that's a dumb question :/
I will try to look over the code again, and read the info you provided.

**Comment by Kirbyrawr** [comment]
No, you don't need to set the network everytime, just at the start or if you are changing to a custom network.

You just need to acces the server.

**Comment by bazooka70** [comment]
> There are two Stellar networks, public and testnet maybe that's your confusion

No. i fully realize that there are both public and testnet.

But the use of a global *static* `Network.Current` in the code makes me think I'm missing something.

Suppose threadTest sets the `Network.UseTestNetwork()` and the threadMain sets `Network.UsePublicNetwork()` before the `threadTest` makes the request to the testnet... `Network.Current` is now set to `PUBLIC` and threadTest is actually using PUBLIC...

Or did I miss anything?

**Comment by fracek** [comment]
I think you have a good understanding of it. In the case of applications like wallets I think it makes sense to have a global object for the current network, but if you're developing something else it's probably better to use the overloaded `Sign` method. You're also right about thread safety (I brought this exact same issue up in #104).

**Comment by Kirbyrawr** [comment]
Thanks @fracek you can explain it better than me!

**Comment by bazooka70** [comment]
@fracek , Thanks. I do think that there is a problem here. We are dealing with both test and main wallets in a very busy web-service environment.

I will look deeper into the code, and might use lock objects (which I really hate to do).
I think @elucidsoft pattern makes the most sense to me e.g.:

`using(var server = Server.Use(Network.Public).Create())`

**Comment by bazooka70** [comment]
@elucidsoft , After i looked at the code, the only place i think should be "fixed" is the `FriendBotRequestBuilder` which uses the `Network.Current`.  
I tried to make a quick pull request:

https://github.com/elucidsoft/dotnet-stellar-sdk/pull/110/commits/d0ab0fce800d8e1a2b78c042557ebff18df13df1

But since the `FriendBotRequestBuilder` checks the `Network.Current` in the constructor (and I removed the tests there), some tests failed. I might try again later.
